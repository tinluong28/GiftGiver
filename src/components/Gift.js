import React, { Component } from "react";
import { Form, FormGroup, FormControl, Button } from "react-bootstrap";

class Gift extends Component {
  constructor() {
    super();
    this.state = {
      person: "",
      present: ""
    };
  }
  render() {
    return (
      <div>
        <Form>
          <Form.Group>
            <Form.Label>Person {this.props.gift.id}</Form.Label>
            <Form.Control
              className="person-input"
              onChange={e => this.setState({ person: e.target.value })}
            />
          </Form.Group>
          <Form.Group>
            <Form.Label>Gift</Form.Label>
            <Form.Control
              className="present-input"
              onChange={e => this.setState({ present: e.target.value })}
            />
          </Form.Group>
        </Form>
        <Button
          className="btn-remove"
          onClick={() => this.props.removeGift(this.props.gift.id)}
        >
          Remove Gift
        </Button>
      </div>
    );
  }
}

export default Gift;
