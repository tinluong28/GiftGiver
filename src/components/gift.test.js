import React from "react";
import { shallow } from "enzyme";
import Gift from "./Gift";

describe("App", () => {
  const mockRemove = jest.fn();
  const id = 1;
  const props = { gift: { id }, removeGift: mockRemove };
  const gift = shallow(<Gift {...props} />);

  it("renders properly", () => {
    expect(gift).toMatchSnapshot();
  });

  it("initialize a person and present in `state`", () => {
    expect(gift.state()).toEqual({ person: "", present: "" });
  });

  describe("when typing into the person input", () => {
    const person = "Uncle";
    beforeEach(() => {
      gift
        .find(".person-input")
        .simulate("change", { target: { value: person } });
    });

    it("update the person in `state`", () => {
      expect(gift.state().person).toEqual(person);
    });
  });

  describe("when typing into the present input", () => {
    const present = "Goft Cub";
    beforeEach(() => {
      gift
        .find(".present-input")
        .simulate("change", { target: { value: present } });
    });

    it("update the present in `state`", () => {
      expect(gift.state().present).toEqual(present);
    });
  });

  describe("when clicking the `Remove Gift`", () => {
    beforeEach(() => {
      gift.find(".btn-remove").simulate("click");
    });

    it("calls the removeGift callback", () => {
      expect(mockRemove).toHaveBeenCalledWith(id);
    });
  });
});
