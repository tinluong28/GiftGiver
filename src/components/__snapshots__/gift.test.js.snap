// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Gift
    gift={
      Object {
        "id": 1,
      }
    }
    removeGift={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ForwardRef(Bootstrap(Form))>
          <ForwardRef(Bootstrap(FormGroup))>
            <ForwardRef(ContextTransform)>
              Person 
              1
            </ForwardRef(ContextTransform)>
            <ForwardRef(ContextTransform)
              className="person-input"
              onChange={[Function]}
            />
          </ForwardRef(Bootstrap(FormGroup))>
          <ForwardRef(Bootstrap(FormGroup))>
            <ForwardRef(ContextTransform)>
              Gift
            </ForwardRef(ContextTransform)>
            <ForwardRef(ContextTransform)
              className="present-input"
              onChange={[Function]}
            />
          </ForwardRef(Bootstrap(FormGroup))>
        </ForwardRef(Bootstrap(Form))>,
        <ForwardRef(Bootstrap(Button))
          className="btn-remove"
          onClick={[Function]}
        >
          Remove Gift
        </ForwardRef(Bootstrap(Button))>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef(Bootstrap(FormGroup))>
              <ForwardRef(ContextTransform)>
                Person 
                1
              </ForwardRef(ContextTransform)>
              <ForwardRef(ContextTransform)
                className="person-input"
                onChange={[Function]}
              />
            </ForwardRef(Bootstrap(FormGroup))>,
            <ForwardRef(Bootstrap(FormGroup))>
              <ForwardRef(ContextTransform)>
                Gift
              </ForwardRef(ContextTransform)>
              <ForwardRef(ContextTransform)
                className="present-input"
                onChange={[Function]}
              />
            </ForwardRef(Bootstrap(FormGroup))>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef(ContextTransform)>
                  Person 
                  1
                </ForwardRef(ContextTransform)>,
                <ForwardRef(ContextTransform)
                  className="person-input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    "Person ",
                    1,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Person ",
                  1,
                ],
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "person-input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Feedback": [Function],
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <ForwardRef(ContextTransform)>
                  Gift
                </ForwardRef(ContextTransform)>,
                <ForwardRef(ContextTransform)
                  className="present-input"
                  onChange={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Gift",
                },
                "ref": null,
                "rendered": "Gift",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "present-input",
                  "onChange": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "Feedback": [Function],
                  "defaultProps": undefined,
                  "propTypes": undefined,
                  "render": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "Check": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Input": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Control": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Feedback": [Function],
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Group": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Label": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Row": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "Text": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Remove Gift",
          "className": "btn-remove",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Remove Gift",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": undefined,
          "propTypes": undefined,
          "render": [Function],
        },
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ForwardRef(Bootstrap(Form))>
            <ForwardRef(Bootstrap(FormGroup))>
              <ForwardRef(ContextTransform)>
                Person 
                1
              </ForwardRef(ContextTransform)>
              <ForwardRef(ContextTransform)
                className="person-input"
                onChange={[Function]}
              />
            </ForwardRef(Bootstrap(FormGroup))>
            <ForwardRef(Bootstrap(FormGroup))>
              <ForwardRef(ContextTransform)>
                Gift
              </ForwardRef(ContextTransform)>
              <ForwardRef(ContextTransform)
                className="present-input"
                onChange={[Function]}
              />
            </ForwardRef(Bootstrap(FormGroup))>
          </ForwardRef(Bootstrap(Form))>,
          <ForwardRef(Bootstrap(Button))
            className="btn-remove"
            onClick={[Function]}
          >
            Remove Gift
          </ForwardRef(Bootstrap(Button))>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef(Bootstrap(FormGroup))>
                <ForwardRef(ContextTransform)>
                  Person 
                  1
                </ForwardRef(ContextTransform)>
                <ForwardRef(ContextTransform)
                  className="person-input"
                  onChange={[Function]}
                />
              </ForwardRef(Bootstrap(FormGroup))>,
              <ForwardRef(Bootstrap(FormGroup))>
                <ForwardRef(ContextTransform)>
                  Gift
                </ForwardRef(ContextTransform)>
                <ForwardRef(ContextTransform)
                  className="present-input"
                  onChange={[Function]}
                />
              </ForwardRef(Bootstrap(FormGroup))>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef(ContextTransform)>
                    Person 
                    1
                  </ForwardRef(ContextTransform)>,
                  <ForwardRef(ContextTransform)
                    className="person-input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": Array [
                      "Person ",
                      1,
                    ],
                  },
                  "ref": null,
                  "rendered": Array [
                    "Person ",
                    1,
                  ],
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": undefined,
                    "propTypes": undefined,
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "person-input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "Feedback": [Function],
                    "defaultProps": undefined,
                    "propTypes": undefined,
                    "render": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <ForwardRef(ContextTransform)>
                    Gift
                  </ForwardRef(ContextTransform)>,
                  <ForwardRef(ContextTransform)
                    className="present-input"
                    onChange={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": "Gift",
                  },
                  "ref": null,
                  "rendered": "Gift",
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "defaultProps": undefined,
                    "propTypes": undefined,
                    "render": [Function],
                  },
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "present-input",
                    "onChange": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": Object {
                    "$$typeof": Symbol(react.forward_ref),
                    "Feedback": [Function],
                    "defaultProps": undefined,
                    "propTypes": undefined,
                    "render": [Function],
                  },
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "Check": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Input": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
              "Label": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": undefined,
                "propTypes": undefined,
                "render": [Function],
              },
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Control": Object {
              "$$typeof": Symbol(react.forward_ref),
              "Feedback": [Function],
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Group": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Label": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Row": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "Text": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "propTypes": undefined,
              "render": [Function],
            },
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Remove Gift",
            "className": "btn-remove",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Remove Gift",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "propTypes": undefined,
            "render": [Function],
          },
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
